tests/integration_tests/security/test_vulnerabilities.py

```diff
@@ -29,309 +25,105 @@
-        guest_kernel_linux_5_10,
-        rootfs,
-        firecracker=firecracker,
-        jailer=jailer,
-        cpu_template=cpu_template,
-    )
-
-
-@pytest.fixture
-def build_microvm_with_custom_template(
-    microvm_factory, guest_kernel_linux_5_10, rootfs, custom_cpu_template
-):
-    """Fixture returning a factory function for microvms with custom cpu templates"""
-    return lambda firecracker=None, jailer=None: configure_microvm(
-        microvm_factory,
-        guest_kernel_linux_5_10,
-        rootfs,
-        firecracker=firecracker,
-        jailer=jailer,
-        custom_cpu_template=custom_cpu_template,
-    )
-
-
-def with_restore(factory, microvm_factory):
-    """Turns the given microvm factory into one that makes the microvm go through a snapshot-restore cycle"""
-
-    def restore(firecracker=None, jailer=None):
-        microvm = factory(firecracker, jailer)
-
-        snapshot = microvm.snapshot_full()
-
-        if firecracker:
-            dst_vm = microvm_factory.build(
-                fc_binary_path=firecracker, jailer_binary_path=jailer
-            )
-        else:
-            dst_vm = microvm_factory.build()
-        dst_vm.spawn()
-        # Restore the destination VM from the snapshot
-        dst_vm.restore_from_snapshot(snapshot, resume=True)
-        dst_vm.cpu_template = microvm.cpu_template
-
-        return dst_vm
-
-    return restore
-
-
-def with_checker(factory, spectre_meltdown_checker):
-    """Turns the given microvm factory function into one that also contains the spectre-meltdown checker script"""
+    def _parse_output(self, output):
+        return {
+            json.dumps(entry)  # dict is unhashable
+            for entry in json.loads(output)
+            if entry["VULNERABLE"]
+        }
 
-    def download_checker(firecracker, jailer):
-        microvm = factory(firecracker, jailer)
-        microvm.ssh.scp_put(spectre_meltdown_checker, REMOTE_CHECKER_PATH)
-        return microvm
+    def get_report_for_guest(self, vm) -> set:
+        """Parses the output of `spectre-meltdown-checker.sh --batch json`
+        and returns the set of issues for which it reported 'Vulnerable'.
 
-    return download_checker
+        Sample stdout:
+        ```
+        [
+          {
+            "NAME": "SPECTRE VARIANT 1",
+            "CVE": "CVE-2017-5753",
+            "VULNERABLE": false,
+            "INFOS": "Mitigation: usercopy/swapgs barriers and __user pointer sanitization"
+          },
+          { ... }
+        ]
+        ```
+        """
+        vm.ssh.scp_put(self.path, REMOTE_CHECKER_PATH)
+        res = vm.ssh.run(REMOTE_CHECKER_COMMAND)
+        return self._parse_output(res.stdout)
+
+    def get_report_for_host(self) -> set:
+        """Runs `spectre-meltdown-checker.sh` in the host and returns the set of
+        issues for which it reported 'Vulnerable'.
+        """
+
+        res = utils.check_output(f"sh {self.path} --batch json")
+        assert res.returncode == 0
```

Comment:
nit: should we assert this in the guest as well?

Reply:
Actually `check_output` should already be checking that, so it's not needed.

Reply:
Fixed